processor/cumulativetodeltaprocessor/internal/tracking/tracker.go

```diff
@@ -81,19 +123,45 @@ func (t *MetricTracker) Convert(in MetricPoint) (out DeltaValue, valid bool) {
 	hashableID := b.String()
 	identityBufferPool.Put(b)
 
+	var keep bool
 	s, ok := t.states.LoadOrStore(hashableID, &State{
 		PrevPoint: metricPoint,
 	})
 	if !ok {
-		return
+		switch t.initialValue {
+		case InitialValueDrop:
+			return
+		case InitialValueAuto:
+			if metricID.StartTimestamp < t.startTime || metricPoint.ObservedTimestamp == metricID.StartTimestamp {
+				return
+			}
+			out.StartTimestamp = metricID.StartTimestamp
+			keep = true
+		case InitialValueKeep:
+			keep = true
+		}
 	}
+
 	valid = true
 
+	if keep {
+		switch metricID.MetricType {
+		case pmetric.MetricTypeHistogram:
+			*out.HistogramValue = metricPoint.HistogramValue.Clone()
+		case pmetric.MetricTypeSum:
+			out.IntValue = metricPoint.IntValue
+			out.FloatValue = metricPoint.FloatValue
+		}
+		return
+	}
```

Comment:
Can we move this inside the `!ok` if condition after the switch statement?

Reply:
done